src/eko/couplings.py

```diff
@@ -265,96 +351,171 @@ def from_dict(cls, theory_card, masses=None):
             hqm_scheme,
         )
 
-    def compute_exact(self, as_ref, nf, scale_from, scale_to):
+    def compute_exact(self, a_ref, nf, scale_from, scale_to):
         """
         Compute via |RGE|.
 
         Parameters
         ----------
-            as_ref: float
-                reference alpha_s
+            as_ref: np array
+                reference alpha_s and alpha
             nf: int
-                value of nf for computing alpha_s
+                value of nf for computing alpha_i
             scale_from: float
                 reference scale
             scale_to : float
                 target scale
 
         Returns
         -------
-            a_s : float
-                strong coupling at target scale :math:`a_s(Q^2)`
+            a : np array
+                couplings at target scale :math:`a(Q^2)`
         """
         # in LO fallback to expanded, as this is the full solution
-        if self.order == 0:
-            return as_expanded(self.order, as_ref, nf, scale_from, float(scale_to))
+        u = np.log(scale_to / scale_from)
+
+        def unidimensional_exact(beta0, b_vec, u, a_ref, method, rtol):
+            def rge(_t, a, b_vec):
+                rge = -(a**2) * (np.sum([a**k * b for k, b in enumerate(b_vec)]))
+                return rge
+
+            res = scipy.integrate.solve_ivp(
+                rge,
+                (0, beta0 * u),
+                (a_ref,),
+                args=[b_vec],
+                method=method,
+                rtol=rtol,
+            )
+            return res.y[0][-1]
+
+        if self.order in [(0, 0), (0, 1), (1, 0), (1, 1)]:
+            return couplings_expanded(
+                self.order, a_ref, nf, scale_from, float(scale_to)
+            )
+        if self.order[0] in [0, 1]:
+            # return expanded solution for a_s and exact for a_em
+            a_s = couplings_expanded(
+                self.order, a_ref, nf, scale_from, float(scale_to)
+            )[0]
+            beta0_qed = beta_qed((0, 2), nf)
+            b_qed_vec = [1.0]
+            # NLO
+            if self.order[1] >= 2:  # I think that at this point this if is always true
```

Comment:
I agree, this should be true at this point

Reply:
actually, why? `order` is coming from the outside ...